File: 21. merge two sorted lists.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function mergeTwoLists(list1: ListNode | null, list2: ListNode | null): ListNode | null {
    let list3: ListNode = [];

    list1.forEach(element1 => {
        list2.forEach(element2 => {
            if(element1 <= element2){
                list3.next = element1;
                continue;
            }
            else{
                list3.next = element2;
            }
        });
    });

    return list3;
};

File: 21. merge two sorted lists.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution
{
    public ListNode MergeTwoLists(ListNode list1, ListNode list2)
    {
        ListNode ListNodeFinal = new ListNode();

        while (list1.val != 0 && list2.val != 0)
        {
            if (list1.val < list2.val)
            {
                ListNodeFinal.val = list1.val;
                list1 = list1.next;
                ListNodeFinal = ListNodeFinal.next;
            }
            else
            {
                ListNodeFinal.val = list2.val;
                list2 = list2.next;
                ListNodeFinal = ListNodeFinal.next;
            }
        }

        return ListNodeFinal;

    }
}